exchanges/binanceus/binanceus.go

```diff
@@ -0,0 +1,1966 @@
+package binanceus
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/thrasher-corp/gocryptotrader/common"
+	"github.com/thrasher-corp/gocryptotrader/common/convert"
+	"github.com/thrasher-corp/gocryptotrader/common/crypto"
+	"github.com/thrasher-corp/gocryptotrader/currency"
+	exchange "github.com/thrasher-corp/gocryptotrader/exchanges"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/asset"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/kline"
+	"github.com/thrasher-corp/gocryptotrader/exchanges/request"
+	"github.com/thrasher-corp/gocryptotrader/log"
+	"github.com/thrasher-corp/gocryptotrader/portfolio/withdraw"
+)
+
+// Binanceus is the overarching type across this package
+type Binanceus struct {
+	validLimits []int
+	exchange.Base
+	obm *orderbookManager
+}
+
+const (
+	// General Data Endpoints
+	serverTime   = "/api/v3/time"
+	systemStatus = "/sapi/v1/system/status"
+
+	// Public endpoints
+	exchangeInfo     = "/api/v3/exchangeInfo"
+	recentTrades     = "/api/v3/trades"
+	aggregatedTrades = "/api/v3/aggTrades"
+	orderBookDepth   = "/api/v3/depth"
+	candleStick      = "/api/v3/klines"
+	tickerPrice      = "/api/v3/ticker/price"
+	averagePrice     = "/api/v3/avgPrice"
+	bestPrice        = "/api/v3/ticker/bookTicker"
+	priceChange      = "/api/v3/ticker/24hr"
+	historicalTrades = "/api/v3/historicalTrades"
+
+	// Withdraw API endpoints
+	tradingStatus = "/sapi/v3/apiTradingStatus"
+	tradeFee      = "/wapi/v3/tradeFee.html"
```

Comment:
While this was probably added while this PR has been up, could you please add the following endpoints with tests? 
https://docs.binance.us/#quick-disable-crypto-withdrawal
and
https://docs.binance.us/#get-user-s-spot-asset-snapshot